karapace/compatibility/__init__.py

```diff
@@ -34,36 +37,62 @@ def is_transitive(self) -> bool:
         return self.value in TRANSITIVE_MODES
 
 
-class IncompatibleSchema(Exception):
-    pass
-
-
-def check_avro_compatibility(reader_schema, writer_schema) -> None:
+def check_avro_compatibility(reader_schema, writer_schema) -> SchemaCompatibilityResult:
     result = AvroChecker().get_compatibility(reader=reader_schema, writer=writer_schema)
     if (
         result.compatibility is SchemaCompatibilityType.incompatible
         and [SchemaIncompatibilityType.missing_enum_symbols] != result.incompatibilities
     ):
-        raise IncompatibleSchema(str(result.compatibility))
+        return result
+
+    return SchemaCompatibilityResult.compatible()
 
 
-def check_compatibility(source: TypedSchema, target: TypedSchema, compatibility_mode: CompatibilityModes) -> None:
+def check_jsonschema_compatibility(reader: Draft7Validator, writer: Draft7Validator) -> SchemaCompatibilityResult:
+    return jsonschema_compatibility(reader, writer)
+
+
+def check_compatibility(
+    source: TypedSchema, target: TypedSchema, compatibility_mode: CompatibilityModes
+) -> SchemaCompatibilityResult:
```

Comment:
Instead of raising, now the error is returned. This allows the API to report the error message.